lib/cell.rb

```diff
@@ -0,0 +1,25 @@
+class Cell
+  attr_reader :coordinate, :ship
+  def initialize(coordinate)
+    @coordinate = coordinate
+    @ship = nil
+    @hit = false
+  end
+
+  def empty?
+    @ship == nil
+  end
+
+  def fired_upon?
+    @hit
+  end
+
+  def fire_upon
+    @hit = true
+    @ship.hit
```

Comment:
I think we need an if statement here. If there is a ship then call hit method. Otherwise it will throw an error when the cell is empty and it is fired upon. We should add a test for that as well.

Reply:
Yes I completely agree.

Reply:
I modified fire_upon so now it looks like this:

```
def fire_upon
    @fired_upon = true
    @ship.hit if @ship
  end
```

I wrote a test for this but it is questionable to me:

```
def test_fire_upon_works_if_there_is_no_ship
  @cell1.fire_upon
end
```

Can we mark this as an issue for later review?

Reply:
Agree with the fire_upon method. 

I think the test should be 

`def test_fire_upon_works_if_there_is_no_ship
  @cell1.fire_upon
  assert_equal true, @cell1.fired_upon?
end`

Reply:
Yes that is a good suggestion I will add that.